### PR TITLE
Allow override of redirect url for authentication in Preview

### PR DESCRIPTION
**Pull request by jamespamplin**
Small change to enable overriding the `redirect_url` used for Google OAuth on Preview (and training-preview).

Passing `redirect-url` as a query parameter to `/login` and `/oauth2callback` will ensure that Google redirects the client to `redirect-url` instead of `/oauth2callback` for authentication and validation.

Required for a proxy to preview in the new mobile viewer: guardian/editorial-viewer#10

### CODE REVIEW

File: standalone/app/controllers/OAuthLoginController.scala [hunk 1]
```diff
@@ -57,13 +57,20 @@ object OAuthLoginController extends Controller with ExecutionContexts with impli
     }
   }
 
+  private def overrideRedirectUrl(config: GoogleAuthConfig)(implicit request: RequestHeader) = {
+    request.getQueryString("redirect-url") match {
+      case Some(url) => config.copy(redirectUrl = url)
```
Comment (theefer) [inline standalone/app/controllers/OAuthLoginController.scala]: In other tools we validated the domain of the redirect url argument. Should we do this here (and everywhere as a rule) for security?

cc @adamnfish

Comment (jamespamplin) [inline standalone/app/controllers/OAuthLoginController.scala]: Yeah we could. The redirect URLs are whitelisted in the Google apps account settings globally, but an extra check on domain here should be fine.

Comment (jamespamplin) [inline standalone/app/controllers/OAuthLoginController.scala]: Fixed.

Comment (theefer) [inline standalone/app/controllers/OAuthLoginController.scala]: :+1: :exclamation:

**Comment by rich-nguyen** [comment]
:+1:

**Comment by jamespamplin** [comment]
Thanks @rich-nguyen!

**Pull request merged**